scripts/test/Muon/fitting_context_test.py

```diff
@@ -86,81 +87,91 @@ class FittingContextTest(unittest.TestCase):
     def setUp(self):
         self.fitting_context = FittingContext()
 
-    def test_len_gives_length_of_fit_list(self):
-        self.assertEqual(0, len(self.fitting_context))
-        self.fitting_context.add_fit(
-            FitInformation(mock.MagicMock(), 'MuonGuassOsc', mock.MagicMock(),
-                           mock.MagicMock()))
-        self.assertEqual(1, len(self.fitting_context))
-
     def test_items_can_be_added_to_fitting_context(self):
         fit_information_object = FitInformation(mock.MagicMock(),
                                                 'MuonGuassOsc',
                                                 mock.MagicMock(),
+                                                mock.MagicMock(),
                                                 mock.MagicMock())
 
+        self.mock_active_fit_history = mock.PropertyMock(return_value=[])
+        type(self.fitting_context).active_fit_history = self.mock_active_fit_history
+        self.fitting_context.all_latest_fits = mock.MagicMock(return_value=[fit_information_object])
+
         self.fitting_context.add_fit(fit_information_object)
 
         self.assertEqual(fit_information_object,
-                         self.fitting_context.fit_list[0])
+                         self.fitting_context.all_latest_fits()[0])
 
     def test_fitfunctions_gives_list_of_unique_function_names(self):
         test_fit_function = 'MuonGuassOsc'
+
+        self.mock_active_fit_history = mock.PropertyMock(return_value=[])
+        type(self.fitting_context).active_fit_history = self.mock_active_fit_history
```

Comment:
I dont understand this. Why do we have the type?

Reply:
We have to mock `@property` getter and setter methods differently because they have the same name, and this is the easiest way